Add documentation for vlan bpf bypass. (#17539)
cilium-team/cilium: Documentation/gettingstarted/kubeproxy-free.rst

Cilium's eBPF kube-proxy replacement by default disallows access to a ClusterIP service from outside the cluster.
This can be allowed by setting ``bpf.lbExternalClusterIP=true``.

> VLAN 802.1q support

Reviewer: The vlan support is not limited to the kube-proxy replacement, e.g., the host firewall feature benefits from it too (/cc @pchaigno to correct me if I'm wrong). Considering that, I think it makes sense to move this section to a documentation page dedicated to the datapath configuration (under the https://docs.cilium.io/en/stable/configuration/).

Author: Yes, you're right I think about benefits for host firewall. But I can't see any age related to datapath configuration - just only rate limiting.

Author: @brb , I've updated PR, but I'm not sure if this is correct...

*******************

> Cilium enables firewall on used native devices and will filter all unknown traffic. VLAN 802.1q packets

Reviewer 2: ```suggestion
Cilium enables firewalling on native devices in use and will filter all unknown traffic. VLAN 802.1q packets
```

> will be always passed through their main device with tag (e.g. vlan device is eth0.4000 and it's main interface is eth0).

Reviewer 2: ```suggestion
will always be passed through their main device and associated tag (e.g. VLAN device is ``eth0.4000`` and its main interface is ``eth0``).
```

Reviewer 2: Adding formatting to `eth0` should also fix the spell check step in the failing GitHub Action.

Author: Resolved, but I've changed 'and' to 'with', cause packets are tagged (with tag) and passed through main device... Sorry, english is not my native language, but somebody had to write something :)

Reviewer 2: No need to apologise, I'm not a native speaker either and I understand. I'm happy to see your contribution, my English might be slightly better so I'm trying to give suggestions -- that's what reviews are for! :)

> By default cilium will allow all tags from used native devices (i.e. if eth0.4000 is controlled by cilium and have

Reviewer 2: ```suggestion
By default, Cilium will allow all tags from the native devices in use (i.e. if ``eth0.4000`` is controlled by Cilium and has
```

> BPF program attached, then vlan tag 4000 will be allowed on eth0 device). Additional vlan tags may be allowed

Reviewer 2: ```suggestion
an eBPF program attached, then VLAN tag ``4000`` will be allowed on device ``eth0``). Additional VLAN tags may be allowed
```

> with cilium-agent flag: ``--vlan-bpf-bypass=4001,4002`` (or helm variable: ``--set bpf.vlan-bpf-bypass=4001,4002``).

Reviewer 2: ```suggestion
with the cilium-agent flag ``--vlan-bpf-bypass=4001,4002`` (or Helm variable ``--set bpf.vlan-bpf-bypass=4001,4002``).
```

> Allowed vlan tags list can't be too big in order to keep BPF program of predictable size. Currently this list

Reviewer 2: ```suggestion
The list of allowed VLAN tags cannot be too big in order to keep an eBPF program of predictable size. Currently this list
```

> should not have more then 5 entries. If you need more then there is only one way for now - you need to allow

Reviewer 2: ```suggestion
should contain no more than 5 entries. If you need more, then there is only one way for now: you need to allow
all tags with cilium-agent flag ``--vlan-bpf-bypass=0``.
```

all tags with cilium-agent flag: ``--vlan-bpf-bypass=0``.

.. note::

>     Currently cilium-agent will scan for available vlan devices / tags only on startup.

Reviewer 2: ```suggestion
    Currently, the cilium-agent will scan for available VLAN devices and tags only on startup.
```
